network/router.go

```diff
@@ -220,12 +226,18 @@ func (r *Router) handleConn(remote *ServerIdentity, c Conn) {
 		if err != nil {
 			if err == ErrTimeout {
 				log.Lvlf5("%s drops %s connection: timeout", r.ServerIdentity.Address, remote.Address)
+				for _, v := range r.errorHandlers {
```

Comment:
can you put that into a small `Router` function please, let's avoid duplication as much as possible.